### ISSUE TITLE
IDISP003 should not warn when assigning with assignment.

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#69]
```cs
namespace RoslynSandbox
{
    using System;

    public sealed class Foo : IDisposable
    {
        private readonly IDisposable disposable;

        public Foo()
        {
            this.DataContext = disposable = new Disposable();
        }

        public object DataContext { get; set; }

        public void Dispose()
        {
            this.disposable?.Dispose();
        }
    }
}
```
